Common++/header/IpUtils.h

```diff
@@ -21,7 +21,8 @@
 
 /// @file
 
-#if defined(WIN32) && !defined(_MSC_VER)
+// Only MinGW doesn't support WinSock2 (not MinGW-w64)
+#if defined(__MINGW32__) && !defined(__MINGW64__)

```

Comment:
ditto

Reply:
https://sourceforge.net/p/predef/wiki/Compilers/

Yes

#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR) ?

Reply:
this should match only mingw-w32 but not mingw-w64_x86_64 or mingw-w64_i686.

What do you think ?

Reply:
What about:
```c++
#if defined(WIN32) && !defined(_MSC_VER) && !defined(__MINGW64__)
```

Reply:
This will not work for MinGW-w64 32 Bit

Reply:
#if defined(WIN32) && !defined(_MSC_VER) && !defined(__MINGW64_MAJOR_VERSION)

Reply:
yes you're right! your idea is better